### DISCUSSION TITLE
First edit does not set dirty flag

### DISCUSSION THREAD

**Discussion by geertjanvdk** [react-hook-form/react-hook-form#10967] [General]
The first edit of my forms never set the the dirty flag: not on form level, or field level. Whether I am using zod or yup, it does not matter.

The 2nd edit, then all is fine and even the first field is considered dirty, which I find very odd..

The only workaround I have now used for months is to do the following in every component using `useForm`, simply calling it once:

```
formMethods.formState.isDirty;
```

For example:
```
  const formMethods = useForm<SchemaOrg>({
    defaultValues: SCHEMA_ORG.getDefault(),
    resolver: yupResolver(SCHEMA_ORG),
  });
```

It doesn't not matter whether I have defaults not. zod or yup, same result (I had to go yup because of other reasons). Schema has `.default("")` on each of the fields. Validation works nicely.

Just wondering why it is.. And whether I have to keep that hack in each component.

**Comment by leapful** [comment]
@geertjanvdk 

Please share a minimal reproducible codesandbox url.
